numpy/core/src/npysort/quicksort.c.src

```diff
@@ -361,3 +361,81 @@ npy_quicksort(void *base, size_t num, size_t size, npy_comparator cmp)
     qsort(base, num, size, cmp);
     return 0;
 }
+
+
+int
+npy_aquicksort(char *v, npy_intp* tosort, npy_intp num, PyArrayObject *arr)
+{
+    npy_intp elsize = PyArray_ITEMSIZE(arr);
+    PyArray_CompareFunc *cmp = PyArray_DESCR(arr)->f->compare;
```

Comment:
For object types, in Python3 it is implemented with two calls to `PyObject_RichCompareBool`, but unless an error is raised it behaves like a compare establishing an ordering. The source is [here](https://github.com/numpy/numpy/blob/maintenance/1.9.x/numpy/core/src/multiarray/arraytypes.c.src#L2590). If an error is raised I think it mistakenly thinks both items compared equal, but the error is caught an raised in `_new_argsortlike`.